File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventsByIdsQueryResponse.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult
import java.util.{ Map => JMap }

/**
 * Events by IDs.
 */
class EventsByIdsQueryResponse extends GraphQLResult[JMap[String, scala.Seq[Event]]] {

    /**
     * Events by IDs.
     */
    def eventsByIds(): scala.Seq[Event] = {
        val data: JMap[String, scala.Seq[Event]] = getData
        if (data != null) data.get(EventsByIdsQueryResponse.OPERATION_NAME) else null
    }

}

object EventsByIdsQueryResponse {

    private final val OPERATION_NAME: String = "eventsByIds"

}

File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventsByIdsQueryResponse.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult

/**
 * Events by IDs.
 */
open class EventsByIdsQueryResponse : GraphQLResult<MutableMap<String, List<Event>>>() {

    companion object {
        const val OPERATION_NAME: String = "eventsByIds"
    }

    /**
     * Events by IDs.
     */
    fun eventsByIds(): List<Event> {
        val data: MutableMap<String, List<Event>> = super.getData()
        return data.getValue(OPERATION_NAME)
    }
}
